lib/puppet/type/mongodb_user.rb

```diff
@@ -18,6 +41,12 @@ def initialize(*args)
     defaultto { @resource[:name] }
   end
 
+  #represents a plaintext password (for mongodb <= 2.4)
+  #only a parameter, since there's no way to compare it with the current value
+  newparam(:password_plain) do
```

Comment:
since it is imposible to compare the plaintext password with the current state (the API returns the hash, even for versions <=2.4) there's no point of making it a property.